handyrl/train.py

```diff
@@ -179,19 +201,22 @@ def compose_losses(policies, values, log_selected_policies, advantages, value_ta
     turn_advantages = advantages.mul(tmasks).sum(-1, keepdim=True)
 
     losses['p'] = (-log_selected_policies * turn_advantages).sum()
-    losses['v'] = ((values - value_targets) ** 2).mul(vmasks).sum() / 2
+    if values is not None:
+        losses['v'] = ((values - value_targets) ** 2).mul(vmasks).sum() / 2
+    if returns is not None:
+        losses['r'] = F.smooth_l1_loss(returns, return_targets, reduction='none').mul(vmasks).sum()
 
     entropy = dist.Categorical(logits=policies).entropy().mul(tmasks.sum(-1))
     losses['ent'] = entropy.sum()
 
-    loss_base = losses['p'] + losses['v']
+    loss_base = losses['p'] + losses.get('v', 0) + losses.get('r', 0)
     losses['total'] = loss_base + entropy.mul(1 - progress * 0.9).sum() * -entropy_regularization
```

Comment:
The coefficient of `progress`, `0.9`, should be set in configuration?
What do you think?

Reply:
agree.
There might be better function than linear one.

Reply:
It's difficult problem because it depends on games.
But I think the adjustment `1 - progress * 0.9` makes sense. So let's just keep this!